### DISCUSSION TITLE
Is there a way to make `ItemsService` not go through API hooks?

### DISCUSSION THREAD

**Discussion by powerseed** [directus/directus#14132] [Archive]
My use case is as follows:

There are two columns in a colleciton, `date` and `time_zone`.

In the collection view (where all records are listed), I want `date` displayed in the time zone specified by `time_zone`.

I think I have to do some processing in a API hook, converting `date` in a `filter('myTable.items.read'), async (input, { keys }, { schema, accountability }) => {...}` method.

But in the collection view, sometimes the `time_zone` value is not included in the `input` param mentioned above (for example, when the user chooses to not include this column in the preset), so I cannot do the conversion between time zones.

To solve it, I think I have to use `ItemsService` to read the complete record, which must include `time_zone`, using some thing like below

```js
const collection = new ItemsService('myTable', { schema, accountability });
let timeZone = collection.readOne(input[0].id).time_zone;
```

Here is the problem:

the `ItemsService.readOne` method itself goes thourgh the API hook, so there is an infinite loop:

1. In collection view, all records are fetched from DB
2. Each record goes thourgh the API hook `filter('myTable.items.read')` 
3. Since no `time_zone` value is found, `ItemsService` reads the full record by its `id`.
4. The `ItemsService.readOne()` method triggers the API hook once again, so it goes back to step 2.

I can work around it, but it is not very clean in my case.

I am wondering if there is simply a way to make `ItemsService` NOT go through the API hooks, like passing in a param `callApiHooks = false`?

Thanks!

**Comment by rijkvanzanten** [comment]
> I am wondering if there is simply a way to make ItemsService NOT go through the API hooks, like passing in a param callApiHooks = false?

Yes*

There's a `emitEvents: false` option you can pass in, however I'm now noticing that's only available for the mutations (eg CrUD) and not for `read` 😱 PR incoming..!

**Reply by rijkvanzanten** [reply]
Ref https://github.com/directus/directus/pull/14164

**Reply by powerseed** [reply]
Got it. Thanks a lot!
